### ISSUE TITLE
[Android] NRE while doing Absolute Navigation from Prism

### ISSUE THREAD

**Issue by zekeinferno** [xamarin/Xamarin.Forms#4403]
### Description
After updating Xamarin Forms, going through a few steps and hitting a line from Prism that does an absolute navigation, the app crashes with an NRE in Android.

I've spoken with the Syncfusion team and it does not look like the issue is on their end. My assumption is that something has changed between how Prism and Xamarin Forms interacts for absolute navigations.

### Steps to Reproduce

1. Click on the carousel item
2. Swipe to the left twice
3. Click any of the 2 buttons on the screen, which both do absolute navigations

### Basic Information

- Version with issue: Xamarin Forms 3.1.0.697729 and beyond
- Last known good version: Xamarin Forms 3.1.0.637273
- IDE: Visual Studio for Mac
- Nuget Packages: Xamarin Forms, Prism and Syncfusion Carousel

### Reproduction Projects
The below project is a cleaned version of my app that is on the last known good version. Updating Xamarin Forms will introduce the issue.
[SyncfusionExample_Working.zip](https://github.com/xamarin/Xamarin.Forms/files/2587549/SyncfusionExample_Working.zip)

The below project is one where I updated Prism and its dependencies to the latest as well as Xamarin Forms to 3.4.0 pre 1 in an attempt to see if the issue has been resolved. Xamarin Forms 3.4.0 pre 2 introduces a different issue with FFImageCaching so I could not test that version.
[SyncfusionExample_Broken.zip](https://github.com/xamarin/Xamarin.Forms/files/2587552/SyncfusionExample_Broken.zip)

Updated the broken project above per request.
[SyncfusionExample_Broken_3.5.0.1006868.zip](https://github.com/xamarin/Xamarin.Forms/files/2696435/SyncfusionExample_Broken_3.5.0.1006868.zip)

**Comment by kingces95** [comment]
Trying to build the busted project results in the following error:

Error occurred while restoring NuGet packages: The local source 'C:\Users\Chris\Downloads\SyncfusionExample_Broken\Example\LocalPackages' doesn't exist.
1>------ Build started: Project: Example.Common, Configuration: Debug Any CPU ------
1>C:\Program Files\dotnet\sdk\2.1.500\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(198,5): error NETSDK1004: Assets file 'C:\Users\Chris\Downloads\SyncfusionExample_Broken\Example\Example.Common\obj\project.assets.json' not found. Run a NuGet package restore to generate this file.
1>Done building project "Example.Common.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

**Comment by zekeinferno** [comment]
@kingces95 My apologies, I forgot to remove a line from the NuGet.config. I have removed the problematic line and re-uploaded both reproduction projects in the original comment.

**Comment by PaulAndreson** [comment]
Hi Team,

Any update on the reported issue?

**Comment by kingces95** [comment]
Wow. That's amazing. 

> 2>obj\Debug\netstandard2.0\Views\Onboarding\PriceMedsTutorialPage.xaml.g.cs(20,39,20,51): error CS0433: The type 'CarouselView' exists in both 'Xamarin.Forms.CarouselView, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' and 'Xamarin.Forms.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=null'

After upgrading to latest XF (3.5.0 or latest nightly), we re-introduced `CarouselView` and that breaks your reproduction. Would you be willing to upgrade to the new CV and re-submit? 

Out of curiosity, have you guys shipped your app using the old `CarouselView`? I assume so. How's it been working for you?

**Comment by kingces95** [comment]
Hm, now I'm getting: 
> 1>MSBUILD : error : Fody: An unhandled exception occurred:
1>MSBUILD : error : Exception:
1>MSBUILD : error : Failed to execute weaver C:\Users\Chris\.nuget\packages\propertychanged.fody\2.6.0\netclassicweaver\PropertyChanged.Fody.dll

**Comment by zekeinferno** [comment]
I have not encountered any weaver-related issues while upgrading the reproduction projects.

The two issues I saw was the one I noted yesterday and Issue #4405 (which can be resolved by updating FFImageLoading). The reproduction project I uploaded yesterday did not encounter the FFImageLoading issue at the version it was on but I did see it at version 4.0.0.

We utilize the PositionSelected event in our actual project but I believe it has been removed from the reproduction project to keep it minimal and straight to the point, it should not have an impact on the NREs (from navigating with Prism and the one in the BottomTabPage’s renderer from trying to upgrade yesterday).

**Comment by zekeinferno** [comment]
There was an issue with the Syncfusion Carousel that I found while updating the reproduction project to XF 3.5.0.979221-nightly 2 weeks prior. The issue seemed to stem from the Syncfusion Carousel itself. I've spoken with Syncfusion themselves and they have given me an updated version of the control that fixes the issue.

I've updated the reproduction project to XF 3.5.0.1006868 and uploaded it above, named "SyncfusionExample_Broken_3.5.0.1006868.zip"
